Development/nmos/control_protocol_state.cpp

```diff
@@ -0,0 +1,360 @@
+#include "nmos/control_protocol_state.h"
+
+#include "nmos/control_protocol_methods.h"
+#include "nmos/control_protocol_resource.h"
+
+namespace nmos
+{
+    namespace experimental
+    {
+        namespace details
+        {
+            // create control class
+            //   where
+            //     properties: vector of NcPropertyDescriptor can be constructed using make_control_class_property
+            //     methods:    vector of NcMethodDescriptor vs assoicated method handler where NcMethodDescriptor can be constructed using make_nc_method_descriptor
+            //     events:     vector of NcEventDescriptor can be constructed using make_nc_event_descriptor
+            control_class make_control_class(const utility::string_t& description, const nc_class_id& class_id, const nc_name& name, const web::json::value& fixed_role, const std::vector<web::json::value>& properties_, const std::vector<nmos::experimental::method>& methods_, const std::vector<web::json::value>& events_)
```

Comment:
Confused by two method signatures  (here and line 34) that are identical apart from the data type of the fixed role. Perhaps a comment to explain when the fixed role should be a JSON value and when a string.

Reply:
Ignore this comment - I realised that one function delegates to the other (DOH!)